pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -486,19 +549,36 @@ func (r *Reconciler) ensureIsInstalled(ctx context.Context, log *zap.SugaredLogg
 		log.Debug("Skipping addon installation as the manifest is empty after parsing")
 		return nil
 	}
-
 	// We delete all resources with this label which are not in the combined manifest
 	selector := labels.SelectorFromSet(r.getAddonLabel(addon))
-	cmd := r.getApplyCommand(ctx, kubeconfigFilename, manifestFilename, selector)
+	cmd := r.getApplyCommand(ctx, kubeconfig, manifest, selector)
 	cmdLog := log.With("cmd", strings.Join(cmd.Args, " "))
 
 	cmdLog.Debug("Applying manifest...")
 	out, err := cmd.CombinedOutput()
-	cmdLog.Debugw("Finished executing command", "output", string(out))
 	if err != nil {
-		return fmt.Errorf("failed to execute '%s' for addon %s of cluster %s: %v\n%s", strings.Join(cmd.Args, " "), addon.Name, cluster.Name, err, string(out))
+		return fmt.Errorf("failed to execute '%s' for addon %s of cluster %s: %v\n%s", strings.Join(cmd.Args, " "), addon.Name, cluster, err, string(out))
+	}
+	cmdLog.Debugw("Finished executing command", "output", string(out))
+	return nil
+}
+
+// ensureIsInstalled ensures proper installation of the addon to appropriate clusters - an addon can have part installed in seed and part in user cluster
+// NOTE: because previously addons were all installed only to user cluster, following code also ensures removal of the seed cluster addon part from the user cluster
+func (r *Reconciler) ensureIsInstalled(ctx context.Context, log *zap.SugaredLogger, addon *kubermaticv1.Addon, cluster *kubermaticv1.Cluster) error {
+	mi := r.setupManifestInteraction(ctx, log, addon, cluster)
+	if mi.err != nil {
+		return mi.err
+	}
+	defer mi.done()
+	var errs []error
+	if err := r.applyManifest(ctx, log, addon, cluster.Name, mi.seedManifestFilename, ""); err != nil {
```

Comment:
for the seed cluster, we don't need to pass a kubeconfig and use in cluster authentication instead